### PR TITLE
Add `id-length` ESLint rule

### PR DESCRIPTION
**Pull request by ehmicky**
This adds the [`id-length`](https://eslint.org/docs/rules/id-length) ESLint rule.
See also https://github.com/netlify/build/pull/1982

Single-letter variable names makes reading code less clear.

### CODE REVIEW

**Review by erezrokah — CHANGES_REQUESTED** [review]
This is huge @ehmicky thanks.
Added a few comments

**Comment by ehmicky** [comment]
Answered this in the PR for Netlify Build, let's solve this first: https://github.com/netlify/build/pull/1982#issuecomment-713112437

File: src/commands/dev/index.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ async function startFrameworkServer({ settings, log, exit }) {
   return ps
 }
 
-const getAddonsUrlsAndAddEnvVariablesToProcessEnv = async ({ api, site, flags }) => {
```
[the remark this answers is not in the document]
Comment (ehmicky) [reply src/commands/dev/index.js]: :+1:

File: src/functions-templates/js/identity-signup/identity-signup.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ exports.handler = async function (event) {
     },
     user_metadata: {
       ...user.user_metadata, // append current user metadata
-      custom_data_from_function: 'hurray this is some extra metadata',
+      custom_data_from_func: 'hurray this is some extra metadata',
```
[the remark this answers is not in the document]
Comment (ehmicky) [reply src/functions-templates/js/identity-signup/identity-signup.js]: Good point :+1:

**Comment by ehmicky** [comment]
Thanks @erezrokah! 
I have removed the `max` constraint.
I also removed all the changes in this PR related to long variable names, so this PR is a little smaller.

**Comment by ehmicky** [comment]
Rebased. Does this look good @erezrokah ?

**Review by erezrokah — APPROVED** [review]

**Pull request merged**